test/unit/sthDatabase_test.js

```diff
@@ -272,6 +272,50 @@ function collectionAccessTests() {
     });
 }
 
+/**
+ * Tests to check that the access to the collections works as expected after reconnect with DB
+ */
+function collectionAccessReconnectTests() {
+    const ORIGINAL_DATA_MODEL = sthConfig.DATA_MODEL;
+    const dataTypes = Object.keys(sthTestConfig.DATA_TYPES);
+    const dataModels = Object.keys(sthConfig.DATA_MODELS);
+
+    dataModels.forEach(function(dataModel) {
+        describe(sthConfig.DATA_MODELS[dataModel] + ' data model', function() {
+            before(function() {
+                sthConfig.DATA_MODEL = sthConfig.DATA_MODELS[dataModel];
+            });
+
+            dataTypes.forEach(function(dataType) {
+                //prettier-ignore
+                it('should notify as error a non-existent ' + sthTestConfig.DATA_TYPES[dataType] + 
+                    ' data collection if it should not be created',
+                    function(done) {
+                        sthDatabase.getCollection(
+                            COLLECTION_NAME_PARAMS,
+                            {
+                                shouldCreate: true,
+                                isAggregated:
+                                    sthTestConfig.DATA_TYPES[dataType] === sthTestConfig.DATA_TYPES.AGGREGATED,
+                                shouldTruncate: false
+                            },
+                            function(err, collection) {
+                                expect(err).to.be(null);
+                                expect(collection).to.be.ok();
```

Comment:
Text expectations should check the response code (500) and the response payload (`DataBase is not connected`)

Reply:
Done in commit https://github.com/telefonicaid/fiware-sth-comet/pull/574/commits/a6b04b689c2bd60b34d4162645468a3578b100c4